---
title: "DADA2_pipeline"
output: html_document
---

```{r}
require(csv)
require(dplyr)
require(tidyr)
```

### Install dada2 library with BiocManager

```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("dada2")
library(dada2)
```

### Define path for folder with reads (in fastq format)

```{r}
path <- "PATH to/reads"
list.files(path)
```

### Install "phyloseq" library for creating DADA2 phyloseq-object 

```{r}
library(phyloseq); packageVersion("phyloseq")
```

### Define forward and reverse fastq files

```{r}
fnFs <- sort(list.files(path, pattern="_R1.fastq", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2.fastq", full.names = TRUE))
```

### Extract sample names:

```{r}
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
```

### Quality of forward reads

```{r}
plotQualityProfile(fnFs[1:2])
```

### Quality of revese reads

```{r}
plotQualityProfile(fnRs[1:2])
```

### Assign the filenames for the filtered fastq.gz files

```{r}
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names
```

### Filtering and trimming reads based on quality of the reads

```{r}
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, trimLeft=c(19,20),truncLen=c(180,180),
              maxN=0,  maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE) # On Windows set multithread=FALSE
```

### Error rates for forward reads

```{r}
errF <- learnErrors(filtFs, multithread=TRUE)
```

```{r}
plotErrors(errF, nominalQ=TRUE)
```

### Error rates for reverse reads

```{r}
errR <- learnErrors(filtRs, multithread=TRUE)
```

```{r}
plotErrors(errR, nominalQ=TRUE)
```

### Sample inference for forward reads

```{r}
dadaFs <- dada(filtFs, err=errF, multithread=TRUE)
```

```{r}
dadaFs[[1]]
```

### Sample inference for reverse reads

```{r}
dadaRs <- dada(filtRs, err=errR, multithread=TRUE)
```

```{r}
dadaRs[[1]]
```

### Merge reads 

```{r}
mergers <- mergePairs(dadaFs, filtFs, dadaRs, filtRs, verbose=TRUE)
```

### Construct an amplicon sequence variant table (ASV) table

```{r}
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
```

```{r}
table(nchar(getSequences(seqtab)))
```

### Remove chimeras

```{r}
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
dim(seqtab.nochim)
```

```{r}
sum(seqtab.nochim)/sum(seqtab)
```

### Report for changing of thr reads from DADA2 pipeline

```{r}
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
head(track)
```

### Assign taxonomy

```{r}
taxa <- assignTaxonomy(seqtab.nochim, "./silva_nr99_v138_train_set.fa.gz", multithread=TRUE, verbose = TRUE)
```

```{r}
taxa.print <- taxa 
rownames(taxa.print) <- NULL
head(taxa.print)
```

### Create a phylogenic tree
```{r}
sequences<-getSequences(seqtab.nochim)
names(sequences)<-sequences
alignment <- AlignSeqs(DNAStringSet(sequences), anchor=NA)
phang.align <- phyDat(as(alignment, "matrix"), type="DNA")
dm <- dist.ml(phang.align)
treeNJ <- NJ(dm)
fit = pml(treeNJ, data=phang.align)
```

```{r}
fitGTR <- update(fit, k=4, inv=0.2)
fitGTR <- optim.pml(fitGTR, model="GTR", optInv=TRUE, optGamma=TRUE,
                    rearrangement = "stochastic", control = pml.control(trace = 0))
```

### Create metadata object

```{r}
sdata <- as.csv("./sample-metadata.csv", sep = ";", row.names = 1, header = TRUE, check.names = TRUE, strindsAsFactor = TRUE)
```

```{r}
sdata <- sdata %>%
  column_to_rownames(var = "SampleID")
sdata$visit <- factor(sdata$visit, labels = c("1","2","3"))
```

```{r}
samdata = sample_data(sdata)
```

### Create a phyloseq object

```{r}
DADA2 <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE),
               tax_table(taxa), sample_data(samdata), phy_tree(fitGTR$tree))
```

### Save this file for future analysis

```{r}
saveRDS(DADA2_ps, file = "DADA2.rds")
```